core/spec/models/spree/order/state_machine_spec.rb

```diff
@@ -30,7 +30,7 @@
 
           it "should not complete the order" do
             expect(order.complete).to be false
-            expect(order.state).to eq("confirm")
+            expect(order.state).to eq("payment")
```

Comment:
If I understood correctly, the order was in the payment state even before, but we were missing an `order.reload` to see that. What is not clear is why we don't need it anymore now. BTW, I think that a preparatory commit, in this case, would help a lot to understand what happens.